backend/dal/Services/Admin/Concrete/BuildingService.cs

```diff
@@ -278,8 +296,18 @@ public override void Remove(Building building)
             building.ThrowIfNotAllowedToEdit(nameof(building), this.User, new[] { Permissions.SystemAdmin, Permissions.AgencyAdmin });
 
             var originalBuilding = this.Context.Buildings.Find(building.Id) ?? throw new KeyNotFoundException();
+            this.Context.Entry(originalBuilding).Collection(p => p.Parcels).Load();
```

Comment:
Don't love the separate requests to load this data, but I don't have time to look into this library https://entityframework.net/ef-plus